ts/store/reducers/wallet/wallets.ts

```diff
@@ -341,12 +343,26 @@ const reducer = (
     case getType(fetchWalletsRequestWithExpBackoff):
     case getType(fetchWalletsRequest):
     case getType(paymentUpdateWalletPsp.request):
+    case getType(updatePaymentStatus.request):
     case getType(deleteWalletRequest):
       return {
         ...state,
         walletById: pot.toLoading(state.walletById)
       };
 
+    case getType(updatePaymentStatus.success):
+      const currentWallets = Object.values(pot.getOrElse(state.walletById, {}));
+      // remove and append the updated wallet
+      const updateWallets = [
+        ...currentWallets
+          .filter(isDefined)
+          .filter(w => w.idWallet !== action.payload.idWallet),
+        action.payload
+      ];
+      return {
+        ...state,
+        walletById: pot.some(toIndexed(updateWallets, _ => _.idWallet))
+      };
```

Comment:
```suggestion
      const currentWallets = pot.getOrElse(state.walletById, {});
      // update the received wallet
      return {
        ...state,
        walletById: pot.some({
          ...currentWallets,
          [action.payload.idWallet]: action.payload
        })
      };
```
Could this be an improvement in order to avoid switching `IndexedById` -> `Array` -> `IndexedById`?

Reply:
Definitely better